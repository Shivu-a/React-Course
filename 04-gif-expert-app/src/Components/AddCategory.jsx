import React, { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onChangedInput = (e) => {
    setInputValue(e.target.value);
  };

  const onSubmit = (e) => {
    // Prevenimos el default
    e.preventDefault();

    // Creamos una variable con el input
    const newInputValue = inputValue.trim();

    // Chequeamos el input
    if (newInputValue.length <= 1) return;

    // Creamos la nueva categoria y reseteamos el value
    onNewCategory(newInputValue);
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Ingrese su busqueda"
        value={inputValue}
        onChange={onChangedInput}
      />
    </form>
  );
};
