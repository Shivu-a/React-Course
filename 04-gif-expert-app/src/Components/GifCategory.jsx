import React from "react";

export const GifCategory = ({ url, id, title }) => {
  return (
    <div className="card">
      <p>{title}</p>
      <img src={url} />
    </div>
  );
};
