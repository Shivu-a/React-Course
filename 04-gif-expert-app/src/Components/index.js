export * from "./AddCategory";
export * from "./GifCategory";
export * from "./GifGrid";
