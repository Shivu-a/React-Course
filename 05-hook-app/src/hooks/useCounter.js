import { useState } from "react";

export const useCounter = (initialValue = 10, value) => {
  const [counter, setCounter] = useState(initialValue);

  const incrementarCounter = (value = 1) => {
    setCounter(counter + value);
  };

  const decrementarCounter = (value = 1) => {
    counter !== 0 && setCounter(counter - value);
  };

  const resetearCounter = () => {
    setCounter(counter - counter);
  };

  return {
    counter,
    incrementarCounter,
    decrementarCounter,
    resetearCounter,
  };
};
