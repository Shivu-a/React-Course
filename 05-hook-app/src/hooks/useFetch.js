import { useEffect, useState } from "react";

export const useFetch = (url) => {
  const [state, setState] = useState({
    data: null,
    isLoading: true,
    error: null,
  });

  const getQuotes = async () => {
    setState({
      ...state,
      isLoading: true,
    });

    const respuesta = await fetch(url);
    const data = await respuesta.json();
    setState({
      ...state,
      data: data,
      isLoading: false,
    });
  };

  useEffect(() => {
    getQuotes();
  }, [url]);

  return state;
};
