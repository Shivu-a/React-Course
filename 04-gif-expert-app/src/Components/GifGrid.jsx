import React from "react";

import { GifCategory } from "./GifCategory";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <div className="gifCategory">
      <h3>{category}</h3>
      {isLoading && <div className="loading"></div>}
      <div className="card-grid">
        {images.map((image) => (
          <GifCategory key={image.id} {...image} />
        ))}
      </div>
    </div>
  );
};
