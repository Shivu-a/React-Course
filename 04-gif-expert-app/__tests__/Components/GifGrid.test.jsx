import { render, screen } from "@testing-library/react";

import { GifGrid } from "../../src/Components/GifGrid";
import { useFetchGifs } from "../../src/hooks/useFetchGifs";
jest.mock("../../src/hooks/useFetchGifs");

describe("Pruebas en GifGrid.jsx", () => {
  test("Deberia hacer match con el snapshot", () => {
    useFetchGifs.mockReturnValue({
      images: [],
      isLoading: true,
    });

    const { container } = render(<GifGrid category={"Rimuru"} />);

    expect(container).toMatchSnapshot();
  });

  test("Deberia de mostrar el loading inicialmente", () => {
    useFetchGifs.mockReturnValue({
      images: [],
      isLoading: true,
    });

    render(<GifGrid category={"Rimuru"} />);

    expect(screen.getByLabelText("div")).toBeTruthy();
  });

  test("Debe de mostrar items cuando se cargan las imagenes", () => {
    const gifs = [
      {
        id: "123",
        title: "Rimuru",
        url: "https://localhost/Rimuru.jpg/",
      },
      {
        id: "321",
        title: "Veldora",
        url: "https://localhost/Veldora.jpg/",
      },
    ];

    useFetchGifs.mockReturnValue({
      images: gifs,
      isLoading: false,
    });

    render(<GifGrid category={"Rimuru"} />);

    expect(screen.getAllByRole("img").length).toBe(gifs.length);
  });
});
