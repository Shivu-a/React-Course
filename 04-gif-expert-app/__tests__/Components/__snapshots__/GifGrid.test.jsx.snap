// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid.jsx Deberia hacer match con el snapshot 1`] = `
<div>
  <div
    class="gifCategory"
  >
    <h3>
      Rimuru
    </h3>
    <div
      aria-label="div"
      class="loading"
    />
    <div
      class="card-grid"
    />
  </div>
</div>
`;
